Refactor sim_time and composable node usage in Collision Monitor

AlexeyMerzlyakov opened the pull request:
This PR fixes Collision Monitor launch-script with:
* Apply to the CM the refactor of `sim_time` usage, introduced in 7703c816a11368fb1a48663dc92e4e93affb637e
* Adding the ability to launch CM as composable node in the container. Enabling this option by-default.
* Adding the support of non-empty root namespace

---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #3131, #2750|
| Primary OS tested on | Ubuntu 20.04 |
| Robotic platform tested on | standard Gazebo TB3 simulation |

---

## Description of documentation updates required from your changes

* No documentation updates required for sim time refactoring.
* Collision Monitor launch-file by-default also launch CM and CM's lifecycle manager in composable container, so all previously described in tutorial commands will remain correct and untouched. Please, let me know, if there an additional notice or sentence about that we have an ability to run w/ and w/o composable container, in tutorial is required.

---

#### For Maintainers: <!-- DO NOT EDIT OR REMOVE -->
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

SteveMacenski commented on nav2_collision_monitor/launch/collision_monitor_node.launch.py:
```diff
@@ -55,43 +60,69 @@ def generate_launch_description():
-    param_substitutions = {
-        'use_sim_time': use_sim_time}
-
-    configured_params = RewrittenYaml(
```
I'm pretty sure this is important (?)

AlexeyMerzlyakov replied:
It was initially removed, since we have no more substitution rewrites in parameters. However, you are right: I've thought twice and checked that `RewrittenYaml` in addtion to parameter rewrites is also making namespace substitution, which was out of support for this script.

Added support the non-root namespace substitution in the https://github.com/ros-planning/navigation2/pull/3604/commits/65c762168955107ed7ad209b8ba6172884379c9c commit.

SteveMacenski commented on nav2_collision_monitor/launch/collision_monitor_node.launch.py:
```diff
@@ -55,43 +60,69 @@ def generate_launch_description():
         description='Full path to the ROS2 parameters file to use for all launched nodes')
 
-    # Create our own temporary YAML files that include substitutions
-    param_substitutions = {
```
Is this not necessary anymore?

AlexeyMerzlyakov replied:
It was used for `use_sim_time` substitution in a `params_file`. Since sim time was re-factored, params substitutions is not necessary anymore.

SteveMacenski reviewed (commented):
Generally approve, but a couple things were deleted that seem unrelated and probably still necessary

AlexeyMerzlyakov commented:
Added support the non-root namespace substitution in the https://github.com/ros-planning/navigation2/pull/3604/commits/65c762168955107ed7ad209b8ba6172884379c9c commit (see https://github.com/ros-planning/navigation2/pull/3604#discussion_r1221820205 comment).
Performed local verificaiton on:

| Namespace / Use composition  | Nav2 composable, CM composable | Nav2 composable, CM usual | Nav2 usual, CM usual |
| ----------- | ----------- | ------- | ------- |
| *none* | PASS | PASS | PASS |
| aaa | PASS | PASS | PASS |

Also, performed local flake8 verification.

Updated the header of the PR with this option with non-root namespaces support.

SteveMacenski commented:
Great! Thanks!
